asyncscope.md

```diff
@@ -625,6 +630,115 @@ class Camera {
 }
 ```
 
+## Recursively spawning work until completion
+Below are three ways you could recursively spawn work on a scope using `let_with_async_scope` or `counting_scope`.
+
+### let_with_async_scope with spawn
+```cpp
+struct tree {
+  std::unique_ptr<tree> left;
+  std::unique_ptr<tree> right;
+  int data;
+};
+
+auto process(ex::scheduler auto sch, ex::counting_scope& scope, tree& t) noexcept {
```

Comment:
This needs to use a template parameter instead of `ex::counting_scope&` because I don't think Anthony's paper specifies the type concretely.
```suggestion
auto process(ex::scheduler auto sch, auto scope, tree& t) noexcept {
```